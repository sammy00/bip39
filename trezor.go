// +build ignore

package main

import (
	"encoding/hex"

	"github.com/sammy00/bip39"
)

func main() {
	trezor := make(map[string][][]string)
	bip39.ReadGoldenJSON(nil, "trezor.json", &trezor)

	//fmt.Printf("%+v\n", trezor)

	// convert to mnemonic goldies to ease testing
	mnemonics := make([]*bip39.Goldie, len(trezor["english"]))
	for i, c := range trezor["english"] {
		entropy, _ := hex.DecodeString(c[0])
		seed, _ := hex.DecodeString(c[2])

		mnemonics[i] = &bip39.Goldie{
			Entropy:  entropy,
			Mnemonic: c[1],
			Seed:     seed,
		}
	}

	if err := bip39.WriteGoldenJSON("trezor.golden", mnemonics); nil != err {
		panic(err)
	}
}
