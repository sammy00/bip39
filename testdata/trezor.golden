[
  {
    "Entropy": "AAAAAAAAAAAAAAAAAAAAAA==",
    "Mnemonic": "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about",
    "Seed": "xVJXw2DAfHICmuvBtTwF7QNiraOOrT4+nvo3COU0lVMfCaaYdZnRgmTB4ckvLPFBYwx6PEq3yBsvABaY50Y7BA=="
  },
  {
    "Entropy": "f39/f39/f39/f39/f39/fw==",
    "Mnemonic": "legal winner thank year wave sausage worth useful legal winner thank yellow",
    "Seed": "LokFgZuHI/4sHRYYYOXuGDAxjb9JqDvUUc+4RAwovW+kV/4SlhBlWaPICTehwQab46OlvTge5iYOjZc5/OH2Bw=="
  },
  {
    "Entropy": "gICAgICAgICAgICAgICAgA==",
    "Mnemonic": "letter advice cage absurd amount doctor acoustic avoid letter advice cage above",
    "Seed": "1x3oVvgaisxl5vyFGjjU1+whb9B5bQpoJ6OtbtVRGjD6KA8S6y5H7SrAO1xGKgNY0Y1p/k+YXsgXeMGzcLZSqA=="
  },
  {
    "Entropy": "/////////////////////w==",
    "Mnemonic": "zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo wrong",
    "Seed": "rCdJVIAiUiIHnXvhgVg3UehvVxAnsEl7W10RIY4KihMzJXKRfw+OWliWIMbxWxHGHe4ydlGhTDThgjEFLkjAaQ=="
  },
  {
    "Entropy": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
    "Mnemonic": "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon agent",
    "Seed": "A1iV8vSBsbDwH8+MKJx5RmCyiZgaePgQZEdwf92WZsoG2lqaVlGBWZt59TuETYpx3Z9DnFKj17Pop5yQashF+g=="
  },
  {
    "Entropy": "f39/f39/f39/f39/f39/f39/f39/f39/",
    "Mnemonic": "legal winner thank year wave sausage worth useful legal winner thank year wave sausage worth useful legal will",
    "Seed": "8rlFCHMry6y8wCD67+z8if6vpmSaVJG4yVLO3klsIUoMezw5LRaHSPLUphK62gdTtSocesU8HpOr1cYyC56V3Q=="
  },
  {
    "Entropy": "gICAgICAgICAgICAgICAgICAgICAgICA",
    "Mnemonic": "letter advice cage absurd amount doctor acoustic avoid letter advice cage absurd amount doctor acoustic avoid letter always",
    "Seed": "EH18AqWqbzjFgIP/dPBMYHwtLA7MVVAdrdctAlt1G8J/6RP/t5b4QcSbHTO2EM8OkdOqI5An9emf5M6eUIjNZQ=="
  },
  {
    "Entropy": "////////////////////////////////",
    "Mnemonic": "zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo when",
    "Seed": "DNbl2Ce7YuuPweJiJUIjgX/QaKdLW0ScwvZnw/H5hadjebQzSNlS4iZbTNEpCQdYs+PCxJEDtQUarC6uuJClKA=="
  },
  {
    "Entropy": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=",
    "Mnemonic": "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon art",
    "Seed": "vahURsaEE3BwkKUgIu3SahyUYilQKfLmDNfE8rvTCXFwr3pNcyRcr6nDzKjVYafD3m9dShC+jtKl5gjWj5L8yA=="
  },
  {
    "Entropy": "f39/f39/f39/f39/f39/f39/f39/f39/f39/f39/f38=",
    "Mnemonic": "legal winner thank year wave sausage worth useful legal winner thank year wave sausage worth useful legal winner thank year wave sausage worth title",
    "Seed": "vAn8oYBPfmnak8LyAo6yOMIn8undowzWNpkjJXhICkAhsUatcX+7fkUc6euDX0NiC/XFFNsPit1J9dEhRJ0+hw=="
  },
  {
    "Entropy": "gICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIA=",
    "Mnemonic": "letter advice cage absurd amount doctor acoustic avoid letter advice cage absurd amount doctor acoustic avoid letter advice cage absurd amount doctor acoustic bless",
    "Seed": "wMUZvQ6Rou1UNX2dHr729a8hihU2JM9PLakRoO2Pegni72GvCsoAcJbfQwAi96K2+5FmGpWJCXBpcg0BXk6YLw=="
  },
  {
    "Entropy": "//////////////////////////////////////////8=",
    "Mnemonic": "zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo zoo vote",
    "Seed": "3UjBBGmMMM/ithQhAySGIvt7sP9pLuuwAImzLSJIThYTkS8KW2lEB76Jn/0x7TmSxFbN9g9dRWS4uj8FppiQrQ=="
  },
  {
    "Entropy": "nohdlSrTYsrrTv40qOkb0g==",
    "Mnemonic": "ozone drill grab fiber curtain grace pudding thank cruise elder eight picnic",
    "Seed": "J03cUlgC98go2O99283FME6HrDU1kTYR+7+phtDJ5UdskWifnIpU/VW9OGBqpqhZWtIT1MnJ+ayj+yFwaaQQKA=="
  },
  {
    "Entropy": "ZhCyWWfNzKnVmHX1y1Cw6nVDMxGGnpML",
    "Mnemonic": "gravity machine north sort system female filter attitude volume fold club stay feature office ecology stable narrow fog",
    "Seed": "Yow4J6iCMpjuaF24T1XKo0tcwZWneOUtRfWbz3Wrpo5NdZDhAdxBS8G71XN2Zvu+810fGQOVO2ZiT5EP7vJFrA=="
  },
  {
    "Entropy": "aKeerKIySHPqzFDLnG7KjMaOpdk2+YeHxgx+vHTmznw=",
    "Mnemonic": "hamster diagram private dutch cause delay private meat slide toddler razor book happy fancy gospel tennis maple dilemma loan word shrug inflict delay length",
    "Seed": "ZMh83n4S7PZwSrlbsUCL7wR8IttMx0kcQnHRcKGyE9ILOFvBWI2cezjxs51BVmW4qQMMnsZT115l+EfY/B/EQA=="
  },
  {
    "Entropy": "wLpajpFBESEPK9Ex89XgjQ==",
    "Mnemonic": "scheme spot photo card baby mountain device kick cradle pact join borrow",
    "Seed": "6nJYlaqujUwc9oLBv9LTWNUu2fDwWRExtVnickuyNPygWqnALFdAfgTuncO0VKpj+/9IOosR3pSWJLnxgxqWEg=="
  },
  {
    "Entropy": "bZvh7m69J6JYEVqtmbcxe5yNKLbXZDHD",
    "Mnemonic": "horn tenant knee talent sponsor spell gate clip pulse soap slush warm silver nephew swap uncle crack brave",
    "Seed": "/VeYKK89odMlRM5Ntcc9U/yKzE3bHjslGjEXnNtx6FPFbS/LEa7TmJjObDSxC1OCdy24eW5Sg3tURorrMSz8PQ=="
  },
  {
    "Entropy": "n2ooeLJSB5mkTvGLx985TnBhoiTSwzzQFbFX10aGmGM=",
    "Mnemonic": "panda eyebrow bullet gorilla call smoke muffin taste mesh discover soft ostrich alcohol speed nation flash devote level hobby quick inner drive ghost inside",
    "Seed": "cr6OBS/EkZ0q3yjVMGtUdLAGnfNbAjA96MFynJU427b8LXMdX4Mhk82ftq7svEaVlKcOPdUIEbUGfzuIsow+jQ=="
  },
  {
    "Entropy": "I9uBYKMdPg3KNojtlBrb8w==",
    "Mnemonic": "cat swing flag economy stadium alone churn speed unique patch report train",
    "Seed": "3rX0VEnmFf7/VkDy5J+TP/UYld47Q4GDKzE5lBxXtZIFpCSAxSF1tu/P+qWKJQOIfB6LNjpwcla90rWHtGVB9Q=="
  },
  {
    "Entropy": "gZekpH8EJfrqpp3uvAXKKcCltcx2zqzA",
    "Mnemonic": "light rule cinnamon wrap drastic word pride squirrel upgrade then income fatal apart sustain crack supply proud access",
    "Seed": "TL3/HKLbgA/WHK5ypXR1/ca6sD5EH9Y/ltq9Hxg+9beCkl8AEF8xgwmn6cPqaWfHgB5GyKWAgmdMhgo3uT7aAg=="
  },
  {
    "Entropy": "Bm3KGiu36KHbKDIUjOmTPuoPOslUjXkxEtmpXJQH760=",
    "Mnemonic": "all hour make first leader extend hole alien behind guard gospel lava path output census museum junior mass reopen famous sing advance salt reform",
    "Seed": "Jul17GREI/SkxPQhXvCbS9fvkk6F0dF8TPPxNsKGPPbfCkdQRWUsV+tftBUTyiotZ3Ird+lUtLP8EfdZBEkZHQ=="
  },
  {
    "Entropy": "8w+MHaZlR49JsAHZTF/EUg==",
    "Mnemonic": "vessel ladder alter error federal sibling chat ability sun glass valve picture",
    "Seed": "KqqSQtqvzuaqnXJp8X1O/iceG5pSkXjX3BOc0YdHCQv51gKV0M50MJp4hSqcqt8K9IquHGJTg5YkB2IkN0vGPw=="
  },
  {
    "Entropy": "wQ7CDcPNn2Usf6wvEjD3o8goOJoUOS8F",
    "Mnemonic": "scissors invite lock maple supreme raw rapid void congress muscle digital elegant little brisk hair mango congress clump",
    "Seed": "e0oQvp2Y5suiZVZtt/E2cY4TmMcctYHhsvRkysHO7fTz4nTcJwADxnCtjQLEVYsvjjnt6id1yeIyx8t5iwaeiA=="
  },
  {
    "Entropy": "9YXBGuxSDbV901PGlVSyGomyD7BlCWb6Cp1vdP2YnY8=",
    "Mnemonic": "void come effort suffer camp survey warrior heavy shoot primary clutch crush open amazing screen patrol group space point ten exist slush involve unfold",
    "Seed": "AfW87VnexI42LyxFtd5ouf1sksZjT0TW1AqraQVlBvDjVSSlGANN3BGS4drNMsHtPqo8OxMciO2OflTEml0JmA=="
  }
]
